modules/labs-react/text-input/lib/TextInput.Label.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import {
+  accessibleHide,
+  createComponent,
+  styled,
+  StyledType,
+  Themeable,
+  useModelContext,
+} from '@workday/canvas-kit-react/common';
+import {type} from '@workday/canvas-kit-react/tokens';
+import {HStack} from '@workday/canvas-kit-labs-react/layout';
+
+import {TextInputModel} from './useTextInputModel';
+import {TextInputModelContext} from './TextInput';
+
+export interface TextInputLabelProps extends Themeable {
+  model?: TextInputModel;
+  /**
+   * If the field is required, provide the title required label.
+   */
+  isRequiredLabel?: string;
+  /**
+   * If true, apply the `accessibleHide` styles to the Label.
+   */
+  isVisuallyHidden?: boolean;
+  /**
+   * The text of the label.
+   */
+  children: React.ReactNode;
+}
+
+const StyledLabel = styled('label')<Pick<TextInputLabelProps, 'isVisuallyHidden'> & StyledType>(
+  type.levels.subtext.large,
+  {
+    fontWeight: type.properties.fontWeights.medium,
+    padding: 0,
+    minWidth: '180px',
+  },
+  ({isVisuallyHidden}) => isVisuallyHidden && accessibleHide
+);
+
+const RequiredAsterisk = styled('abbr')<Pick<TextInputLabelProps, 'theme'> & StyledType>(
+  {
+    fontSize: type.properties.fontSizes[16],
+    fontWeight: type.properties.fontWeights.regular,
+    textDecoration: 'unset',
+  },
+  ({theme}) => {
+    return {
+      color: theme.canvas.palette.error.main,
+    };
+  }
+);
+
+export const TextInputLabel = createComponent('label')({
+  displayName: 'TextInput.Label',
+  Component: (
+    {isRequiredLabel, isVisuallyHidden, theme, model, children, ...elemProps}: TextInputLabelProps,
+    ref
+  ) => {
+    const {state} = useModelContext(TextInputModelContext, model);
+
+    return (
+      <StyledLabel
```

Comment:
Kinda wanted this to be a `<Box as='label' ...` but that doesn't seem to be allowed when using styled:

```
Type 'HTMLDivElement' is missing the following properties from type 'HTMLLabelElement': control, form, htmlForts(2769)
```

https://codesandbox.io/s/admiring-feistel-u601g?file=/src/App.tsx

Reply:
`const StyledBox = styled(Box as typeof TextInputLabel)` fixes it, but is a bit weird to have to use

Reply:
Swapped to using css